src/io_threads.c

```diff
@@ -554,3 +555,56 @@ void trySendPollJobToIOThreads(void) {
     aeSetPollProtect(server.el, 1);
     IOJobQueue_push(jq, IOThreadPoll, server.el);
 }
+
+void setTLSNegotiationCallback(void (*cb)(void *)) {
+    tls_negotiation_cb = cb;
+}
+
+static void ioThreadTLSNegotiation(void *data) {
+    client *c = (client *)data;
+    tls_negotiation_cb(c->conn);
+    c->io_read_state = CLIENT_COMPLETED_IO;
+}
+
+/*
+ * This function attempts to offload TLS negotiation for a client connection to an I/O thread.
+ * Returns C_OK if the TLS negotiation was successfully queued for processing by an I/O thread,
+ * or C_ERR if the client is not eligible for offloading.
+ * Parameters:
+ *   conn: The connection object for which TLS negotiation should be performed
+ */
+int trySendTLSNegotiationToIOThreads(connection *conn) {
+    if (server.io_threads_num <= 1) {
+        return C_ERR;
+    }
+
+    if (!(conn->flags & CONN_FLAG_CLIENT)) {
+        return C_ERR;
+    }
+
+    client *c = connGetPrivateData(conn);
+    if (c->io_read_state != CLIENT_IDLE) {
+        return C_OK;
+    }
+
+    if (server.active_io_threads_num <= 1) {
+        return C_ERR;
+    }
```

Comment:
Consider a scenario where the main thread sends a job to the IO thread. After the IO thread completes the job, the main thread deactivates all threads. In this case, we want to ensure the main thread processes the returned job from the IO thread before performing an accept operation even if the io threads are not active.

To achieve this:

First, check if IO threads are enabled at all (less expensive check)
Then, verify the read_state is not idle
Finally, check the active state

Unlike read/write where we anyway check for the read/write states, the accept flow operates at the connection layer rather than the client layer, so we can't check the read state there.